DBZ-2333 Modularize and edit content for PostgreSQL connector

TovaCohen opened the pull request:
This is a major revision of the doc for the PostgreSQL connector.  The updates include:

- Comments that contain “Category”, “Type”, “ModuleID” and “Title” are annotations for splitting the upstream file into multiple downstream files. This lets us have one file upstream, and, as required, multiple assembly and module files downstream.
- Content revisions for consistency with Red Hat doc style conventions: 
   - Edits for clarity and completeness that focus on user tasks.
   - Present tense and not future tense. 
   - Active voice and not passive voice. 
   - No contractions or Latin phrases (i.e., e. g., via)
   - Shorter and complete sentences. 
   - Procedures have prerequisites.
   - Tables have headings. 
   - Headings use sentence case and not title case
- For downstream, sections that have many subsections have a minitoc for faster navigation. (Upstream, the right-side panel provides this.)
- There are a few instances of `endif` statements immediately followed by `ifdef` statements with the same qualifier, for example: 

   > endif::community[] 
   > ifdef::community[]

   This is intentional for addressing a problem that occurs when the upstream file is split. We’re looking for a better solution, but for now, this doesn’t hurt, though it might not help either. 

It's important for these updates to have a careful review. After this content is updated and merged, I will use this organization as a template for reorganizing the doc for the other connectors. Also, where content is the same for different connectors, I will copy that content from this updated file to the other files.  

To help you see the organization, outlines for before and after organization for both upstream and downstream are here: 
https://docs.google.com/document/d/1FEtKHZH0KLcjwpPOmkr7mAGpzeHkfnOpfg01wnFfR6k/edit?usp=sharing

Let me know what needs to change.

Naros commented on documentation/modules/ROOT/pages/connectors/postgresql.adoc:
```diff
@@ -12,356 +14,156 @@ ifdef::community[]
-<3> Tells the server to allow `<youruser>` on `localhost` to receive replication changes using `IPV6`
+[id="snapshot-mode-settings"]
+.Settings for `snapshot.mode` connector configuration property
+[cols="20%a,80%a",options="header"]
```
We began the move away from using `options="header"` for tables and I think we should continue that here for consistency across all of the documentation.  There are multiple occurrences of this in this file btw.

TovaCohen replied:
In a table, `options="header" is required downstream because it improves accessibility for visually impaired folks who are hearing the text from a screen reader. 
Why are you moving away from using it?
And can you reverse that decision?

Naros replied:
I believe this was a decision that came down when the MySQL documentation was first reworked.  We were told that the use of that was redundant and I've since stopped using it when adding new tables to the documentation.  If they should be retained, then we need to review existing documentation for all `[cols=` occurrences and make sure that the options block is restored.

TovaCohen replied:
I can do that - Should I go ahead and create a DBZ Jira to update doc files so that all tables have the header option?

Naros replied:
Sure; especially if its critical for downstream in someway.  It's something we can also work toward over time too if its not something that necessarily needs to all be done at once.

TovaCohen commented on documentation/modules/ROOT/pages/connectors/postgresql.adoc:
```diff
@@ -12,356 +14,156 @@ ifdef::community[]
 endif::community[]
 
-{prodname}'s PostgreSQL Connector can monitor and record row-level changes in the schemas of a PostgreSQL database.
+{prodname}'s PostgreSQL connector captures row-level changes in the schemas of a PostgreSQL database. PostgreSQL versions 10, 11, and 12 are supported. 
```
Fixed in the next commit. 
I duplicated the paragraph and conditionalized and updated the one for community and conditionalized the original for product.

TovaCohen commented on documentation/modules/ROOT/pages/connectors/postgresql.adoc:
```diff
@@ -12,356 +14,156 @@ ifdef::community[]
 == Overview
 
-PostgreSQL's https://www.postgresql.org/docs/current/static/logicaldecoding-explanation.html[_logical decoding_] feature was first introduced in version 9.4 and is a mechanism which allows the extraction of the changes which were committed to the transaction log and the processing of these changes in a user-friendly manner via the help of an https://www.postgresql.org/docs/current/static/logicald ...
+PostgreSQL's link:https://www.postgresql.org/docs/current/static/logicaldecoding-explanation.html[_logical decoding_] feature was introduced in version 9.4. It is a mechanism that allows the extraction of the changes that were committed to the transaction log and the processing of these changes in a user-friendly manner with the help of an link:https://www.postgresql.org/docs/current/static/logic ...
```
Of course. I might not have understood the implication when I started working on this content, but I certainly knew this by the end. I have update all content so that it is clear that installation of a plug-in might be required only upstream. Downstream, there is no longer content related to installing a plug-in. So this is fixed in the next commit. (subject of course to your review/approval :-)

TovaCohen commented on documentation/modules/ROOT/pages/connectors/postgresql.adoc:
```diff
@@ -12,356 +14,156 @@ ifdef::community[]
 endif::community[]
 
-The connector then produces a _change event_ for every row-level insert, update, and delete operation that was received, recording all the change events for each table in a separate Kafka topic. Your client applications read the Kafka topics that correspond to the database tables they're interested in following, and react to every row-level event it sees in those topics.
+The connector produces a _change event_ for every row-level insert, update, and delete operation that was captured and sends change event records for each table in a separate Kafka topic. Your client applications read the Kafka topics that correspond to the database tables of interest, and can react to every row-level event they receive from those topics.
```
Done in the next commit.

TovaCohen commented on documentation/modules/ROOT/pages/connectors/postgresql.adoc:
```diff
@@ -12,356 +14,156 @@ ifdef::community[]
+PostgreSQL normally purges write-ahead log (WAL) segments after some period of time. This means that the connector does not have the complete history of all changes that have been made to the database. Therefore, when the PostgreSQL connector first connects to a particular PostgreSQL database, it starts by performing a _consistent snapshot_ of each of the database schemas. After the connector com ...
 
-The connector is also tolerant of failures. As the connector reads changes and produces events, it records the position in the write-ahead log with each event. If the connector stops for any reason (including communication failures, network problems, or crashes), upon restart it simply continues reading the WAL where it last left off. This includes snapshots: if the snapshot was not completed whe ...
+The connector is tolerant of failures. As the connector reads changes and produces events, it records the position in the WAL for each event. If the connector stops for any reason (including communication failures, network problems, or crashes), upon restart the connector continues reading the WAL where it last left off. This includes snapshots. If the connector stops during a snapshot, the conne ...
```
Updated as you suggest in the next commit.

TovaCohen commented on documentation/modules/ROOT/pages/connectors/postgresql.adoc:
```diff
@@ -12,356 +14,156 @@ ifdef::community[]
+=== Custom snapshotter SPI
 
-For more advanced usages, the user can provide an implementation of the `io.debezium.connector.postgresql.spi.Snapshotter` interface. This interfaces allows control of most of the aspects of how snapshots operate, such as whether to take a snapshot or not and the way the options used to open the snapshot transaction or take locks.
+For more advanced uses, you can provide an implementation of the `io.debezium.connector.postgresql.spi.Snapshotter` interface. This interaces allows control of most of the aspects of how the connector performs snapshots. This includes whether or not to take a snapshot, the options for opening the snapshot transaction, and whether to take locks. 
```
Fixed in the next commit.

TovaCohen commented on documentation/modules/ROOT/pages/connectors/postgresql.adoc:
```diff
@@ -1169,27 +1055,27 @@ The _semantic type_ describes how the Kafka Connect schema captures the _meaning
 |`STRING`
 |n/a
-|Contains the string representation of a timestamp range with (local system) time zone.
+|Contains the string representation of a timestamp range with the local system  time zone.
```
Fixed in the next commit.

TovaCohen commented on documentation/modules/ROOT/pages/connectors/postgresql.adoc:
```diff
@@ -1419,47 +1324,32 @@ When `hstore.handling.mode` configuration property is set to `json` (the default
-When a column is defined using a domain type that extends one of the default database types and the domain type defines a custom length/scale, the generated schema will inherit that defined length/scale.
-
-When a column is defined using a domain type that extends another domain type that defines a custom length/scale, the generated schema will **not** inherit the defined length/scale because the PostgreSQL driver's column metadata implementation.
+When a column is defined to contain a domain type that extends another domain type that defines a custom length or scale, the generated schema does *not* inherit the defined length oe scale because of the PostgreSQL driver's column metadata implementation.
```
Fixed in the next commit.

TovaCohen commented on documentation/modules/ROOT/pages/connectors/postgresql.adoc:
```diff
@@ -1419,47 +1324,32 @@ When `hstore.handling.mode` configuration property is set to `json` (the default
+=== Network address types
 
-PostgreSQL also have data types that can store IPv4, IPv6, and MAC addresses. It is better to use these instead of plain text types to store network addresses, because these types offer input error checking and specialized operators and functions.
+PostgreSQL haddata types that can store IPv4, IPv6, and MAC addresses. It is better to use these types instead of plain text types to store network addresses. Network address types offer input error checking and specialized operators and functions.
```
Fixed in the next commit.

TovaCohen commented on documentation/modules/ROOT/pages/connectors/postgresql.adoc:
```diff
@@ -1728,36 +2021,40 @@ The following configuration properties are _required_ unless a default value is
+
+endif::community[]
 ifdef::product[]
 The only supported value is `pgoutput`.
```
No. Ouch. I should have caught this. 
Downstream, is the default for `plugin.name` - `pgoutput`?

TovaCohen commented:
@Naros The most recent commit contains updates based on your review, with two outstanding issues: 
-- What is the downstream default for the `plugin.name` property? Is it `pgoutput`?
-- In tables, the "header" option is required downstream. Can it remain in upstream content?

Thanks again for the speedy review. I Really Appreciate IT!

TovaCohen commented:
@Naros  I just rebased my topic branch against master. 
After rebasing, it looked like the extra space in "a timestamp range with the local system  time" was already removed. 
I removed the extraneous "J". 
In the section about the `plugin.name` property, I added a sentence, for downstream-only, that says you must explicitly set it to `pgoutput`. 
I committed and pushed my updates. 
Let me know if anything else is needed.

TovaCohen commented:
Thanks @Naros !  I'm really glad to hear that you think this is an improvement! And I'm glad that you confirmed that no needed content was omitted. 
The details are all important - and getting the details just right makes the customer experience much better 
I'm planning to apply the same organization to the doc for the other connectors, as I get to them. 
Thanks again for the quick turnaround on this!
